Use Lambda for readability

kssumin opened the pull request:
## Description
<!-- 
Please describe what this PR do.
 -->
Closes #936

### Summary
I found using Anonymous inner classes  [src/main/java/oss/fosslight/service/impl/SelfCheckServiceImpl.java](https://github.com/fosslight/fosslight/blob/develop/src/main/java/oss/fosslight/service/impl/SelfCheckServiceImpl.java)
Before Java 8, the only way to partially support closures in Java was by using anonymous inner classes. Java 8 introduced lambdas, which are significantly more readable and should be used instead.

### AS-IS
```java
		Collections.sort(srcList, new Comparator<OssComponents>() {
			@Override
			public int compare(OssComponents oc1, OssComponents oc2) {
				return oc1.getOssName().toUpperCase().compareTo(oc2.getOssName().toUpperCase());
			}
		});
```
### TO-BE
```java
		Collections.sort(srcList,
				(oc1, oc2) -> oc1.getOssName().toUpperCase().compareTo(oc2.getOssName().toUpperCase()));
```

## Type of change
<!--
Please insert 'x' one of the type of change.
 -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

70825 commented:
@jongwooo @kssumin 
I think it would be nice if @kssumin looked up where to sort in other files as well as the current ones, and modified them all, what do you think?

kssumin commented:
> @jongwooo @kssumin I think it would be nice if @kssumin looked up where to sort in other files as well as the current ones, and modified them all, what do you think?

Yes!  I'll apply that issue to all files!

I have one question. That example the second parameter of sort (i.e. the Comparator part) is too long, so I moved it to the next line. I wonder if there are any rules for this part.

70825 commented:
@kssumin 
> That example the second parameter of sort (i.e. the Comparator part) is too long, so I moved it to the next line. I wonder if there are any rules for this part.

In general, it is right to move on to the next line.

When you complete the code modification, modify the issue and PR contents.

jongwooo commented on src/main/java/oss/fosslight/service/impl/VerificationServiceImpl.java:
```diff
@@ -973,18 +973,14 @@ public Map<String, Object> processVerification(Map<Object, Object> map, T2File f
-							return arg1.compareTo(arg2);
-						}
-					};
+				Collections.sort(readmePathList, (arg1, arg2) -> {
+					if (arg1.split("\\/").length > arg2.split("\\/").length) {
+						return 1;
+					} else if (arg1.split("\\/").length < arg2.split("\\/").length) {
+						return -1;
+					} else {
+						return arg1.compareTo(arg2);
+					}
```
@kssumin Since each condition includes a `return`, I think the `else` statement  is unnecessary. WDYT?

```suggestion
				Collections.sort(readmePathList, (arg1, arg2) -> {
					if (arg1.split("\\/").length > arg2.split("\\/").length) {
						return 1;
					}
					if (arg1.split("\\/").length < arg2.split("\\/").length) {
						return -1;
					}
					return arg1.compareTo(arg2);

```

kssumin replied:
Oh, Thank you!

jongwooo commented on src/main/java/oss/fosslight/service/impl/VerificationServiceImpl.java:
```diff
@@ -2379,14 +2373,11 @@ public List<OssComponents> setMergeGridData(List<OssComponents> gridData) {
+			Collections.sort(tempData, (o1, o2) -> {
+				if (o1.getLicenseName().length() >= o2.getLicenseName().length()) {
+					return 1;
+				}else {
```
```suggestion
				} else {

```

70825 reviewed (commented):
Can you please modify the `VerificationServiceImpl.java` file again?

soimkim commented:
@Min-Kyungsun , Please review this PR.
